### DISCUSSION TITLE
Try to avoid explicitly passing context around

### DISCUSSION THREAD

**Discussion by williamwjs** [graphql-java/graphql-java#3105] [Q&A]
I am currently storing my context in the `GraphQlContext`. To be able to access this context, right now I have to pass either this `GraphQlContext` or `DataFetchingEnvironment` along the code path to make sure code down the road would be able to retrieve the information from the context.

Question is: Is there a way that we could implicitly pass this around, like through the `RequestContextHolder` or something similar?
(I tried to store the context inside `RequestContextHolder`, but sometimes it would result in "No thread-bound request found")

**Comment by bbakerman** [comment]
What framework are you running in?  Off hand I dont know what `RequestContextHolder` is?

**Reply by williamwjs** [reply]
`RequestContextHolder` is from Spring: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/context/request/RequestContextHolder.html

**Reply by dondonz** [reply]
Hi @williamwjs,

For Spring for GraphQL questions, could you please post on their GitHub repo https://github.com/spring-projects/spring-graphql/issues or alternatively, post on StackOverflow with the `spring-graphql` tag https://stackoverflow.com/questions/tagged/spring-graphql. The Spring for GraphQL maintainers monitor both.

This repo is for the GraphQL Java engine itself (without any framework).
